datafusion/src/optimizer/single_distinct_to_groupby.rs

```diff
@@ -62,6 +63,10 @@ fn optimize(plan: &LogicalPlan) -> Result<LogicalPlan> {
             group_expr,
         }) => {
             if is_single_distinct_agg(plan) {
+                if is_max_or_min(plan) {
```

Comment:
I don't think this optimization is limited to plans with a single distinct aggregate (the way it is for count). 

In other words, I think it is possible to *always* rewrite `min(distinct x)` to `min(x) as "min(distinct x)"` and `max(distinct x)` to `max(x) as "max(distinct x)"`

Reply:
Great point, thanks!❤